apollo-router/src/services/layers/persisted_queries/manifest_poller.rs

```diff
@@ -343,15 +355,35 @@ impl PersistedQueryManifestPoller {
         }
     }
 
-    pub(crate) fn get_operation_body(&self, persisted_query_id: &str) -> Option<String> {
+    pub(crate) fn get_operation_body(
+        &self,
+        persisted_query_id: &str,
+        client_name: Option<String>,
+    ) -> Option<String> {
         let state = self
             .state
             .read()
             .expect("could not acquire read lock on persisted query manifest state");
-        state
+        if let Some(body) = state
             .persisted_query_manifest
-            .get(persisted_query_id)
+            .get(&FullPersistedQueryOperationId {
+                operation_id: persisted_query_id.to_string(),
+                client_name: client_name.clone(),
+            })
             .cloned()
+        {
+            Some(body)
+        } else if client_name.is_some() {
```

Comment:
Do we need this else if? It looks like we would be returning the null operation for the a request for an operation with a client_name. If `client_name` is Some and nothing exists in `state.persisted_query_manifest` for it, `None` feels like the right return?

Reply:
What this is saying is — if we don't have an operation that is specifically designated for this client name, try one that was published without a client name.  (If we didn't have this, all PQs would suddenly stop working for any clients that send a client name as soon as you upgraded to this version.)

The reason it's a conditional at all instead of just an "else try `client_name: None`" is because if `client_name` is None, it would be making the same exact call twice, which seems wasteful.

I should add comments to this function though!

Reply:
Ooohhhh riiight! that totally makes sense! comment will definitely help, but logic is definitely good! thanks :)